homeassistant/util/temperature.py

```diff
@@ -27,8 +27,8 @@ def convert(temperature: float, from_unit: str, to_unit: str) -> float:
                                                              TEMPERATURE))
 
     if from_unit == to_unit:
-        return temperature
```

Comment:
It should not be the responsibility of a util method to round values as it is not aware of where it is going to be used. It should be rounded in places where we care about a rounded value, ie when we set the state. (So in `entity.py` where we do the conversion.)